src/hapi/pipelines/database/food_price.py

```diff
@@ -0,0 +1,116 @@
+from .currency import Currency
+from .metadata import Metadata
+from .wfp_commodity import WFPCommodity
+from .wfp_market import WFPMarket
+
+logger = getLogger(__name__)
+
+
+class FoodPrice(BaseUploader):
+    def __init__(
+        self,
+        session: Session,
+        datasetinfo: Dict[str, str],
+        countryiso3s: List[str],
+        metadata: Metadata,
+        currency: Currency,
+        commodity: WFPCommodity,
+        market: WFPMarket,
+    ):
+        super().__init__(session)
+        self._datasetinfo = datasetinfo
+        self._countryiso3s = countryiso3s
+        self._metadata = metadata
+        self._currency = currency
+        self._commodity = commodity
+        self._market = market
+
+    def populate(self):
+        logger.info("Populating WFP price table")
+        reader = Read.get_reader("hdx")
+        headers, iterator = reader.read(datasetinfo=self._datasetinfo)
+        datasetinfos = []
+        for country in iterator:
+            countryiso3 = country["countryiso3"]
+            if countryiso3 not in self._countryiso3s:
+                continue
+            dataset_name = country["url"].split("/")[-1]
+            datasetinfos.append(
+                {
+                    "dataset": dataset_name,
+                    "format": "csv",
+                    "admin_single": countryiso3,
+                }
+            )
+        warnings = set()
+        errors = set()
+        for datasetinfo in datasetinfos:
+            headers, rows = reader.read(datasetinfo)
+            hapi_dataset_metadata = datasetinfo["hapi_dataset_metadata"]
+            hapi_resource_metadata = datasetinfo["hapi_resource_metadata"]
+            self._metadata.add_hapi_metadata(
+                hapi_dataset_metadata, hapi_resource_metadata
+            )
+            countryiso3 = datasetinfo["admin_single"]
+            dataset_name = hapi_dataset_metadata["hdx_stub"]
+            resource_id = hapi_resource_metadata["hdx_id"]
+            for row in rows:
+                market = row["market"]
+                if "#" in market:
+                    continue
+                market_code = self._market.get_market_code(countryiso3, market)
+                if not market_code:
+                    add_missing_value_message(
+                        errors, dataset_name, "market code", market
+                    )
+                    continue
+                commodity_code = self._commodity.get_commodity_code(
+                    row["commodity"]
+                )
+                currency_code = row["currency"]
+                unit = row["unit"]
+                price_flag = row["priceflag"]
+                price_type = row["pricetype"]
+                price = row["price"]
+                reference_period_start = parse_date(
+                    row["date"], date_format="%Y-%m-%d"
+                )
+                reference_period_end = reference_period_start + relativedelta(
```

Comment:
Can you explain this?

Reply:
The food price reference period is one month. I added a comment.